src/move/emojicoin_arena/sources/emojicoin_arena.move

```diff
@@ -0,0 +1,1037 @@
+        self: &mut Melee, address: address
+    ) {
+        add_if_not_contains(&mut self.all_entrants, address);
+    }
+
+    inline fun melee_available_rewards_decrement(
+        self: &mut Melee, amount: u64
+    ) {
+        self.available_rewards = self.available_rewards - amount;
+    }
+
+    inline fun melee_available_rewards_increment(
+        self: &mut Melee, amount: u64
+    ) {
+        self.available_rewards = self.available_rewards + amount;
+    }
+
+    inline fun melee_emojicoin_0_locked_decrement(
+        self: &mut Melee, amount: u64
+    ) {
+        aggregator_v2::sub(&mut self.emojicoin_0_locked, amount);
+    }
+
+    inline fun melee_emojicoin_0_locked_increment(
+        self: &mut Melee, amount: u64
+    ) {
+        aggregator_v2::add(&mut self.emojicoin_0_locked, amount);
+    }
+
+    inline fun melee_emojicoin_1_locked_decrement(
+        self: &mut Melee, amount: u64
+    ) {
+        aggregator_v2::sub(&mut self.emojicoin_1_locked, amount);
+    }
+
+    inline fun melee_emojicoin_1_locked_increment(
+        self: &mut Melee, amount: u64
+    ) {
+        aggregator_v2::add(&mut self.emojicoin_1_locked, amount);
+    }
+
+    inline fun melee_exited_entrants_add_if_not_contains(
+        self: &mut Melee, address: address
+    ) {
+        add_if_not_contains(&mut self.exited_entrants, address);
+    }
+
+    inline fun melee_exited_entrants_remove_if_contains(
+        self: &mut Melee, address: address
+    ) {
+        remove_if_contains(&mut self.exited_entrants, address);
+    }
+
+    inline fun melee_locked_in_entrants_add_if_not_contains(
+        self: &mut Melee, address: address
+    ) {
+        add_if_not_contains(&mut self.locked_in_entrants, address);
+    }
+
+    inline fun melee_locked_in_entrants_remove_if_contains(
+        self: &mut Melee, address: address
+    ) {
+        remove_if_contains(&mut self.locked_in_entrants, address);
+    }
+
+    inline fun melee_n_swaps_increment(self: &mut Melee) {
+        aggregator_v2::add(&mut self.n_swaps, 1);
+    }
+
+    inline fun melee_swaps_volume_increment(
+        self: &mut Melee, amount: u128
+    ) {
+        aggregator_v2::add(&mut self.swaps_volume, amount);
+    }
+
+    /// Accepts a mutable reference to avoid freezing references up the stack.
+    inline fun next_melee_market_ids(registry_ref_mut: &mut Registry): vector<u64> {
+        let n_markets = get_n_registered_markets();
+        let market_ids;
+        loop {
+            let market_id_0 = random_market_id(n_markets);
+            let market_id_1 = random_market_id(n_markets);
+            if (market_id_0 == market_id_1) continue;
+            market_ids = sort_unique_market_ids(market_id_0, market_id_1);
+            if (!registry_ref_mut.melee_ids_by_market_ids.contains(market_ids))
+                break;
+        };
+        market_ids
+
+    }
+
+    /// Pseudo-random proxy for `random_market_id` for use during `init_module`.
```

Comment:
```suggestion
    /// Pseudo-random substitute for `random_market_id`, since the Aptos randomness API
    /// is not available during `init_module`.
```

Just to make it consistent with the others, sorry for the nit. The code is just so good I don't have much else to contribute! 🍾